pkg/apisix/cluster.go

```diff
@@ -535,12 +547,14 @@ func (c *cluster) listResource(ctx context.Context, url string) (*listResponse,
 	return &list, nil
 }
 
-func (c *cluster) createResource(ctx context.Context, url string, body io.Reader) (*createResponse, error) {
+func (c *cluster) createResource(ctx context.Context, url, resource string, body io.Reader) (*createResponse, error) {
 	req, err := http.NewRequestWithContext(ctx, http.MethodPut, url, body)
 	if err != nil {
 		return nil, err
 	}
+	start := time.Now()
 	resp, err := c.do(req)
+	c.metricsCollector.RecordAPISIXLatency(time.Now().Sub(start))
 	if err != nil {
 		return nil, err
```

Comment:
I think maybe here should add another metrics to record the error ratio.

Reply:
- name: `request_error_total`(apisixRequestErrors) & `request_error_ratio`
- type: counter & guage
- calculation: ` apisixRequestErrorRadio := c.metricsCollector.apisixRequestErrors / c.metricsCollector.apisixRequests`
- labels: resource
the metric should be like this?

Reply:
Yes, but the current PR is quite large, let's implement this in another PR.